### PR TITLE
Update sawtooth-core to use Transact

### PR DESCRIPTION
**Pull request by agunde406**
Added changes journal FFI updates as well for block validator changes.

### CODE REVIEW

**Comment by agunde406** [comment]
> This pull request **introduces 6 alerts** when merging [4df3186](https://github.com/hyperledger/sawtooth-core/commit/4df3186b231379a4c3a5023b7a2bd51af6d90b15) into [cf06014](https://github.com/hyperledger/sawtooth-core/commit/cf06014beb3a3f241a89ac73d8ce67eaf40b9b48) - [view on LGTM.com](https://lgtm.com/projects/g/hyperledger/sawtooth-core/rev/pr-571be4725fdcf3d12ff6761f0608eb10deeaa858)
> 
> **new alerts:**
> 
>     * 6 for Unused import

These have been fixed

File: validator/sawtooth_validator/journal/journal.py [hunk 1]
```diff
@@ -38,15 +38,23 @@ def __init__(
         fork_cache_keep_time=300,  # seconds
         data_dir=None,
         observers=None,
+        genesis_observers=None,
```
Comment (ltseeley) [inline validator/sawtooth_validator/journal/journal.py:41]: Typo in commit message: s/Cylinder/Cylinder signer/

File: validator/src/journal/journal_ffi.rs [hunk 2]
```diff
@@ -67,11 +72,28 @@ use py_object_wrapper::PyObjectWrapper;
...
+            Ok(bool) => bool,
+            Err(err) => {
+                error!(
+                    "Unable to check if the genesis block should be built: {}",
+                    err
+                );
+                return ErrorCode::Unknown;
+            }
+        } {
+            if let Err(err) = self.genesis_controller.start() {
```
Comment (ltseeley) [inline validator/src/journal/journal_ffi.rs:90]: Would be a bit simpler if you moved this to an `Ok(true)` arm of the match above

File: validator/sawtooth_validator/journal/journal.py (hunk 1 above)
Comment (agunde406) [reply validator/sawtooth_validator/journal/journal.py:41]: Fixed

File: validator/src/journal/journal_ffi.rs (hunk 2 above)
Comment (agunde406) [reply validator/src/journal/journal_ffi.rs]: updated

File: validator/src/journal/journal_ffi.rs [hunk 3]
```diff
@@ -68,11 +73,30 @@ use py_object_wrapper::PyObjectWrapper;
...
 
 impl Journal {
-    fn start(&mut self) {
+    fn start(&mut self) -> ErrorCode {
+        match self.genesis_controller.requires_genesis() {
+            Ok(create_genesis) => {
+                if create_genesis {
+                    if let Err(err) = self.genesis_controller.start() {
+                        error!("Unable to create genesis block: {}", err);
+                        return ErrorCode::Unknown;
```
Comment (peterschwarz) [inline validator/src/journal/journal_ffi.rs]: These should be genesis specific errors. When this fails, it logs badly.

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: validator/src/journal/journal_ffi.rs (hunk 3 above)
Comment (agunde406) [reply validator/src/journal/journal_ffi.rs]: That was updated in a later commit

File: validator/Cargo.toml [hunk 4]
```diff
@@ -6,15 +6,16 @@ authors = ["Intel Corporation"]
...
 lazy_static = "1.0"
 log = { version = "0.4", features = ["std"] }
 libc = ">=0.2.35"
 metrics = { version = "0.12", features = ["std"] }
 protobuf = "2.0"
 python3-sys = "0.2"
-sawtooth = { version = "0.5", features = ["validator-internals"] }
-transact = "0.2"
+transact = {version = "0.3", features = ["sawtooth-compat"]}
+sawtooth = { version = "0.6", features = ["validator-internals"] }
```
Comment (ltseeley) [inline validator/Cargo.toml:17]: This should be kept in alphabetical order. Can you add spaces around the content of the braces for transact for consistency, too?

File: validator/src/journal/journal_ffi.rs [hunk 5]
```diff
@@ -248,11 +293,47 @@ pub unsafe extern "C" fn journal_new(
...
+            context_manager.clone(),
+        ))))
+        .with_signer(identity_signer.clone())
+        .with_state_view_factory(state_view_factory.clone())
+        .start()
+    {
+        Ok(publisher) => publisher,
+        Err(err) => {
+            error!("Unable to start block publisher: {}", err);
+            return ErrorCode::Unknown;
```
Comment (ltseeley) [inline validator/src/journal/journal_ffi.rs:319]: I'll update these to return `CreateError` instead

Comment (ltseeley) [reply validator/src/journal/journal_ffi.rs]: Fixed

File: validator/Cargo.toml (hunk 4 above)
Comment (agunde406) [reply validator/Cargo.toml]: fixed

**Review by vaporos — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]
Tests successfully run locally

**Pull request merged**